src/main/java/io/vertxbeans/ContextRunnerImpl.java

```diff
@@ -57,10 +58,19 @@ public ContextRunnerImpl(Vertx vertx) {
     }
 
 
+
+
     private <T> Consumer<Handler<AsyncResult<T>>>  wrap(Consumer<Handler<AsyncResult<T>>> consumer){
-        Context context = vertx.getOrCreateContext();
-        return resultHandler -> context.runOnContext(
-                v->consumer.accept(result -> context.runOnContext(v1 ->resultHandler.handle(result))));
+        Context context;
+        if (vertx instanceof VertxImpl) {
+            // Create a new context
+            context = ((VertxImpl) vertx).createEventLoopContext();
+        } else {
+            // Use the default vert.x interface behaviour of getOrCreateContext() which will return the same context
+            context = vertx.getOrCreateContext();
+        }
```

Comment:
```suggestion
        Context context = ((VertxInternal) vertx).createEventLoopContext();
```
`VertxInternal` seems a slightly better choice than `VertxImpl`.
Also, my guess is I chose `getOrCreateContext` because it was guaranteed to create a new context when called from within the main thread (because the main thread isn't a Vertx thread). This hack has caught up with us now that the context always gets stored on the current thread, regardless of what kind of thread it is.
Since we always want to create a new context, let's do that.
I don't think `Vertx` is supposed to be extended by applications, so I'm not too worried about getting a `ClassCastException`. And, in any case, I'd rather see that than have this method silently call `getOrCreateContext`, which may or may not do what we want it to.

Reply:
Thank you very much for your review Rob. I agree with you. I have made that change and pushed another commit.